ios/browser/api/ai_chat/ai_chat_delegate.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_IOS_BROWSER_API_AI_CHAT_AI_CHAT_DELEGATE_H_
+#define BRAVE_IOS_BROWSER_API_AI_CHAT_AI_CHAT_DELEGATE_H_
+
+#import <Foundation/Foundation.h>
+#import "ai_chat.mojom.objc.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+OBJC_EXPORT
+@protocol AIChatDelegate <NSObject>
+- (nullable NSString*)getPageTitle;
+- (nullable NSURL*)getLastCommittedURL;
+- (void)getPageContentWithCompletion:(void (^)(NSString* _Nullable content,
```

Comment:
why is this method named `WithCompletion`? We generally try to keep the same method names across chromium/ios

Reply:
This is an iOS thing: https://developer.apple.com/documentation/swift/calling-objective-c-apis-asynchronously and doesn't have anything to do with Chromium.

Chromium also does the same thing: https://source.chromium.org/chromium/chromium/src/+/main:ios/chrome/browser/ui/authentication/authentication_flow.mm;l=170